dominode-bootstrapper/dominode_bootstrapper/geonodeadmin.py

```diff
@@ -175,20 +175,58 @@ def from_geonode_manager(cls, geonode_manager: GeoNodeManager):
         )
 
     def list_workspaces(self):
-        raise NotImplementedError  # TODO: Provide implementation
-
-    def create_workspace(self):
-        raise NotImplementedError  # TODO: Provide implementation
+        response = self.client.get(
+            f'{self.base_url}/rest/workspaces',
+            auth=(DEFAULT_USERNAME, DEFAULT_PASSWORD)
+        )
+        return response.json()
```

Comment:
Not directly related to this method, but you need to modify `GeoServerManager.__init__()` in order to accept admin user credentials. 

These are most likely not the same as the GeoNode admin credentials (at least on a standard install they are not, it is possible that we change them to be the same). 

Do not implicitly reuse `DEFAULT_USERNAME` and `DEFAULT_PASSWORD` - these are meant as a helper for initialization. If you want to, you may rename these to `DEFAULT_GEONODE_ADMIN_USERNAME` and `DEFAULT_GEONODE_ADMIN_PASSWORD` and add respective `DEFAULT_GEOSERVER_ADMIN_USERNAME` and `DEFAULT_GEOSERVER_ADMIN_PASSWORD`, but don't rely on them implicitly, allow setting them during initialization of the manager instance